launchers/junit/src/testFixtures/java/org/eclipse/dataspaceconnector/junit/launcher/EdcRuntimeExtension.java

```diff
@@ -0,0 +1,184 @@
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.stream.Stream;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.eclipse.dataspaceconnector.boot.system.ExtensionLoader.loadMonitor;
+
+/**
+ * A JUnit extension for running an embedded EDC runtime as part of a test fixture.
+ * A custom gradle task printClasspath is used to determine the runtime classpath of the module to run.
+ * The runtime obtains a classpath determined by the Gradle build.
+ * <p>
+ * This extension attaches a EDC runtime to the {@link BeforeTestExecutionCallback} and {@link AfterTestExecutionCallback} lifecycle hooks. Parameter injection of runtime services is supported.
+ */
+public class EdcRuntimeExtension extends EdcExtension {
+    final String moduleName;
+    final Map<String, String> properties;
+    private Thread runtimeThread;
+    private static final String GRADLE_WRAPPER_UNIX = "gradlew";
+    private static final String GRADLE_WRAPPER_WINDOWS = "gradlew.bat";
+    private static final String GRADLE_WRAPPER;
+    private static final Monitor MONITOR = loadMonitor();
+
+    static {
+        GRADLE_WRAPPER = (System.getProperty("os.name").toLowerCase().contains("win")) ? GRADLE_WRAPPER_WINDOWS : GRADLE_WRAPPER_UNIX;
+    }
+
+    public EdcRuntimeExtension(String moduleName, Map<String, String> properties) {
+        this.moduleName = moduleName;
+        this.properties = Map.copyOf(properties);
+    }
+
+    @Override
+    public void beforeTestExecution(ExtensionContext extensionContext) throws Exception {
+        // Find the project root directory, moving up the directory tree
+        var root = findRoot(new File(".").getCanonicalFile());
+        if (root == null) {
+            throw new EdcException("Could not find " + GRADLE_WRAPPER + " in parent directories.");
+        }
+
+        // Run a Gradle custom task to determine the runtime classpath of the module to run
+        String[] command = {
+                new File(root, GRADLE_WRAPPER).getCanonicalPath(),
+                "-q",
+                moduleName + ":printClasspath"
+        };
+        Process exec = Runtime.getRuntime().exec(command);
+        InputStream inputStream = exec.getInputStream();
+        var classpathString = new String(inputStream.readAllBytes());
+        if (exec.waitFor() != 0) {
+            throw new EdcException("Failed to run gradle command: " + String.join(" ", command));
```

Comment:
Should be nice to include in the exception the value of `classpathString`, because I wasn't able to run tests from IDEA because it wasn't able to find my `java` command, and I had to debug to understand which was the problem. Reporting the output could help resolving such issues.

Reply:
Implemented (also outputting stderr)